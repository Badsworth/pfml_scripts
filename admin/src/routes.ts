const routes: {
  [key: string]: string;
} = {
  dashboard: "/",
  features: "/features",
  help: "/help",
  login: "/login",
  maintenance: "/maintenance",
  settings: "/settings",
  users: "/users",
};

export default routes;
