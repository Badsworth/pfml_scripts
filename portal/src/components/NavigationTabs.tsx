import Link from "next/link";
import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

const NavigationTabs = ({ activePath, tabs }) => {
  const isActive = (tab) => tab.href === activePath;

  return (
    <nav className="border-bottom border-base-lighter margin-bottom-5 display-flex width-full">
      {tabs.map((tab) => (
        <Link key={tab.href} href={tab.href}>
          <a
            aria-current={isActive(tab) ? "page" : undefined}
            className={classNames(
              "display-inline-block padding-y-2 margin-bottom-neg-1px text-no-underline width-card",
              {
                // inactive page link
                "text-base-dark hover:text-primary hover:text-underline":
                  !isActive(tab),
                // active page link
                "border-primary border-bottom-05 text-bold text-primary":
                  isActive(tab),
              }
            )}
          >
            {tab.label}
          </a>
        </Link>
      ))}
    </nav>
  );
};

NavigationTabs.propTypes = {
  activePath: PropTypes.string.isRequired,
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      href: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default NavigationTabs;
