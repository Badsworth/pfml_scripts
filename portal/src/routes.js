/* eslint sort-keys: ["error", "asc"] */
/**
 * @file Routes to various pages in the application are defined here rather than being
 * hard-coded into various files.
 */
const routes = {
  api: {
    applications: "/applications",
    claims: "/claims",
    employers: "/employers",
    users: "/users",
  },
  applications: {
    address: "/applications/address",
    bondingLeaveAttestation: "/applications/bonding-leave-attestation",
    caringLeaveAttestation: "/applications/caring-leave-attestation",
    checklist: "/applications/checklist",
    concurrentLeaves: "/applications/concurrent-leaves",
    concurrentLeavesDetails: "/applications/concurrent-leaves-details",
    concurrentLeavesIntro: "/applications/concurrent-leaves-intro",
    dateOfBirth: "/applications/date-of-birth",
    dateOfChild: "/applications/date-of-child",
    employerBenefits: "/applications/employer-benefits",
    employerBenefitsDetails: "/applications/employer-benefits-details",
    employerBenefitsIntro: "/applications/employer-benefits-intro",
    employmentStatus: "/applications/employment-status",
    familyMemberDateOfBirth: "/applications/family-member-date-of-birth",
    familyMemberName: "/applications/family-member-name",
    familyMemberRelationship: "/applications/family-member-relationship",
    gender: "/applications/gender",
    getReady: "/applications/get-ready",
    index: "/applications",
    intermittentFrequency: "/applications/intermittent-frequency",
    leavePeriodContinuous: "/applications/leave-period-continuous",
    leavePeriodIntermittent: "/applications/leave-period-intermittent",
    leavePeriodReducedSchedule: "/applications/leave-period-reduced-schedule",
    leaveReason: "/applications/leave-reason",
    name: "/applications/name",
    notifiedEmployer: "/applications/notified-employer",
    otherIncomes: "/applications/other-incomes",
    otherIncomesDetails: "/applications/other-incomes-details",
    paymentMethod: "/applications/payment-method",
    phoneNumber: "/applications/phone-number",
    previousLeavesIntro: "/applications/previous-leaves-intro",
    previousLeavesOtherReason: "/applications/previous-leaves-other-reason",
    previousLeavesOtherReasonDetails:
      "/applications/previous-leaves-other-reason-details",
    previousLeavesSameReason: "/applications/previous-leaves-same-reason",
    previousLeavesSameReasonDetails:
      "/applications/previous-leaves-same-reason-details",
    reasonPregnancy: "/applications/reason-pregnancy",
    reducedLeaveSchedule: "/applications/reduced-leave-schedule",
    review: "/applications/review",
    scheduleFixed: "/applications/schedule-fixed",
    scheduleRotating: "/applications/schedule-rotating",
    scheduleRotatingNumberWeeks: "/applications/schedule-rotating-number-weeks",
    scheduleVariable: "/applications/schedule-variable",
    ssn: "/applications/ssn",
    start: "/applications/start",
    stateId: "/applications/state-id",
    status: "/applications/status",
    success: "/applications/success",
    // For routes that don't have a page to point to yet, we can route them
    // to a placeholder page. This allows us to search our code for routes.applications.todo,
    // which is less confusing than seeing routes.applications.checklist
    todo: "/applications/checklist",
    uploadCertification: "/applications/upload-certification",
    uploadDocsOptions: "/applications/upload-docs-options",
    uploadId: "/applications/upload-id",
    workPatternType: "/applications/work-pattern-type",
  },
  auth: {
    createAccount: "/create-account",
    forgotPassword: "/forgot-password",
    login: "/login",
    resetPassword: "/reset-password",
    verifyAccount: "/verify-account",
  },
  employers: {
    addOrganization: "/employers/organizations/add-organization",
    cannotVerify: "/employers/organizations/cannot-verify",
    confirmation: "/employers/applications/confirmation",
    createAccount: "/employers/create-account",
    dashboard: "/employers/dashboard",
    finishAccountSetup: "/employers/finish-account-setup",
    newApplication: "/employers/applications/new-application",
    organizations: "/employers/organizations",
    review: "/employers/applications/review",
    status: "/employers/applications/status",
    success: "/employers/applications/success",
    verificationSuccess: "/employers/organizations/success",
    verifyContributions: "/employers/organizations/verify-contributions",
    welcome: "/employers/welcome",
  },
  external: {
    inLocoParentis:
      "https://www.dol.gov/agencies/whd/fact-sheets/28C-fmla-eldercare",
    massTaxConnect: "https://mtc.dor.state.ma.us/mtc/_/",
    massgov: {
      benefitsGuide:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefits-guide",
      benefitsGuide_aboutBondingLeave:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefits-guide#about-family-leave-to-bond-with-a-child-",
      benefitsGuide_aboutCaringLeave:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefits-guide#about-family-leave-to-care-for-a-family-member-",
      benefitsGuide_aboutMedicalLeave:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefits-guide#about-medical-leave-",
      benefitsGuide_seriousHealthCondition:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefits-guide#what-is-a-serious-health-condition?-",
      calculateHours: "https://www.mass.gov/pfml-calculate-hours",
      calculateReductions:
        "https://www.mass.gov/guides/what-to-expect-when-you-apply-for-paid-family-and-medical-leave-benefits#-calculate-any-reductions-",
      caregiverCertificationForm: "https://www.mass.gov/family-caring-form",
      caregiverRelationship:
        "https://www.mass.gov/family-caring-leave-relationships",
      caseCreationErrorGuide:
        "https://www.mass.gov/forms/apply-for-paid-leave-if-you-received-an-error",
      consentAgreement:
        "https://www.mass.gov/info-details/massachusetts-department-of-family-and-medical-leave-informed-consent-agreement",
      dfml: "https://www.mass.gov/orgs/department-of-family-and-medical-leave",
      dfmlRegulations:
        "https://www.mass.gov/regulations/458-CMR-200-family-and-medical-leave",
      employerAccount:
        "https://www.mass.gov/how-to/creating-an-employer-account-to-review-paid-family-and-medical-leave-pfml-applications",
      employerReimbursements:
        "https://www.mass.gov/forms/request-reimbursement-for-your-employees-paid-leave-benefits",
      employersGuide:
        "https://www.mass.gov/info-details/employer-paid-leave-application-review-guide",
      federalEmployerIdNumber:
        "https://www.mass.gov/info-details/finding-your-employers-federal-employer-identification-number-fein",
      feedbackClaimant: "https://www.mass.gov/paidleave-claimant-feedback",
      feedbackEmployer: "https://www.mass.gov/paidleave-employer-feedback",
      healthcareProviderForm: "https://www.mass.gov/medical-leave-form",
      howToApplyPaidLeave:
        "https://www.mass.gov/guides/what-is-paid-family-and-medical-leave#-how-can-i-apply-for-paid-leave-massachusetts-benefits",
      identityProof:
        "https://www.mass.gov/info-details/identity-proof-for-paid-leave",
      index:
        "https://www.mass.gov/topics/paid-family-and-medical-leave-in-massachusetts",
      informedConsent: "https://www.mass.gov/paidleave-informedconsent",
      intermittentLeaveGuide:
        "https://www.mass.gov/intermittent-leave-instructions",
      mailFaxInstructions: "https://www.mass.gov/pfmlsubmitinfo",
      medicalBonding: "https://www.mass.gov/pfml-medical-bonding",
      multipleBirths: "https://www.mass.gov/pfml-multiple-births",
      paidLeave:
        "https://www.mass.gov/topics/paid-family-and-medical-leave-in-massachusetts",
      pfml: "https://www.mass.gov/pfml",
      preparingToVerifyEmployer:
        "https://www.mass.gov/info-details/preparing-to-verify-your-employer-paid-family-and-medical-leave-account",
      privacyPolicy: "https://www.mass.gov/privacypolicy",
      reductionsEmployerBenefits:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefit-reductions#earned-time-off-through-your-employer-",
      reductionsOverview:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefit-reductions",
      requestAnAppealForPFML:
        "https://www.mass.gov/how-to/appealing-a-denied-paid-family-or-medical-leave-claim",
      schedulingLeaveGuide:
        "https://www.mass.gov/info-details/intermittent-and-reduced-leave-schedules",
      taxLiability: "https://www.mass.gov/pfml-tax-liability",
      verifyEmployer: "https://www.mass.gov/pfml-verify-employer",
      whenCanIUsePFML:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-fact-sheet#when-can-i-use-pfml?-",
      whichPaidLeave: "https://www.mass.gov/which-paid-leave",
      zeroBalanceEmployer: "https://www.mass.gov/pfml-zero-balance-employer",
    },
    puertoRicanBirthCertificate: "https://prfaa.pr.gov/faqs",
    workVisa:
      "https://travel.state.gov/content/travel/en/us-visas/employment/temporary-worker-visas.html",
  },
  index: "/",
  user: {
    consentToDataSharing: "/user/consent-to-data-sharing",
    convert: "/user/convert-to-employer",
  },
};

/**
 * @param {string} url - path, with or without query param
 * @returns {boolean}
 */
export const isEmployersRoute = (url) => {
  return Object.values(routes.employers).includes(getRouteFromUrl(url));
};

/**
 * @param {string} url - path, with or without query param
 * @returns {boolean}
 */
export const isApplicationsRoute = (url) => {
  return Object.values(routes.applications).includes(getRouteFromUrl(url));
};

/**
 * @param {string} url - path, with or without query param
 * @returns {string}
 */
const getRouteFromUrl = (url) => {
  let route = url;
  const queryStringIndex = route.indexOf("?");

  if (queryStringIndex !== -1) {
    route = route.substring(0, queryStringIndex);
  }

  if (route.endsWith("/")) {
    route = route.substring(0, route.length - 1);
  }

  return route;
};

export default routes;
