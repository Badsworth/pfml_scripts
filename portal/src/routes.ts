/* eslint sort-keys: ["error", "asc"] */
/**
 * @file Routes to various pages in the application are defined here rather than being
 * hard-coded into various files.
 */
const routes = {
  api: {
    applications: "/applications",
    claims: "/claims",
    employers: "/employers",
    payments: "/payments",
    users: "/users",
  },
  applications: {
    address: "/applications/address",
    bondingLeaveAttestation: "/applications/bonding-leave-attestation",
    caringLeaveAttestation: "/applications/caring-leave-attestation",
    checklist: "/applications/checklist",
    concurrentLeaves: "/applications/concurrent-leaves",
    concurrentLeavesDetails: "/applications/concurrent-leaves-details",
    concurrentLeavesIntro: "/applications/concurrent-leaves-intro",
    dateOfBirth: "/applications/date-of-birth",
    dateOfChild: "/applications/date-of-child",
    department: "/applications/department",
    employerBenefits: "/applications/employer-benefits",
    employerBenefitsDetails: "/applications/employer-benefits-details",
    employerBenefitsIntro: "/applications/employer-benefits-intro",
    employmentStatus: "/applications/employment-status",
    familyMemberDateOfBirth: "/applications/family-member-date-of-birth",
    familyMemberName: "/applications/family-member-name",
    familyMemberRelationship: "/applications/family-member-relationship",
    find: "/applications/find",
    gender: "/applications/gender",
    getReady: "/applications/get-ready",
    index: "/applications",
    intermittentFrequency: "/applications/intermittent-frequency",
    leavePeriodContinuous: "/applications/leave-period-continuous",
    leavePeriodIntermittent: "/applications/leave-period-intermittent",
    leavePeriodReducedSchedule: "/applications/leave-period-reduced-schedule",
    leaveReason: "/applications/leave-reason",
    name: "/applications/name",
    notifiedEmployer: "/applications/notified-employer",
    otherIncomes: "/applications/other-incomes",
    otherIncomesDetails: "/applications/other-incomes-details",
    paymentMethod: "/applications/payment-method",
    phoneNumber: "/applications/phone-number",
    previousLeavesIntro: "/applications/previous-leaves-intro",
    previousLeavesOtherReason: "/applications/previous-leaves-other-reason",
    previousLeavesOtherReasonDetails:
      "/applications/previous-leaves-other-reason-details",
    previousLeavesSameReason: "/applications/previous-leaves-same-reason",
    previousLeavesSameReasonDetails:
      "/applications/previous-leaves-same-reason-details",
    reasonPregnancy: "/applications/reason-pregnancy",
    reducedLeaveSchedule: "/applications/reduced-leave-schedule",
    review: "/applications/review",
    scheduleFixed: "/applications/schedule-fixed",
    scheduleRotating: "/applications/schedule-rotating",
    scheduleRotatingNumberWeeks: "/applications/schedule-rotating-number-weeks",
    scheduleVariable: "/applications/schedule-variable",
    ssn: "/applications/ssn",
    start: "/applications/start",
    stateId: "/applications/state-id",
    status: {
      claim: "/applications/status",
      payments: "/applications/status/payments",
    },
    // Note that this route is used by the back-end when generating notification emails for
    // the claimant (see notifications.py). Changes here will need to be coordinated with the back-end.
    success: "/applications/success",
    taxWithholding: "/applications/tax-withholding",
    // For routes that don't have a page to point to yet, we can route them
    // to a placeholder page. This allows us to search our code for routes.applications.todo,
    // which is less confusing than seeing routes.applications.checklist
    todo: "/applications/checklist",
    upload: {
      bondingProofOfBirth: "/applications/upload/proof-of-birth",
      bondingProofOfPlacement: "/applications/upload/proof-of-placement",
      caringCertification:
        "/applications/upload/family-member-medical-certification",
      index: "/applications/upload",
      medicalCertification: "/applications/upload/medical-certification",
      otherId: "/applications/upload/other-id",
      pregnancyCertification:
        "/applications/upload/pregnancy-medical-certification",
      stateId: "/applications/upload/state-id",
    },
    uploadCertification: "/applications/upload-certification",
    uploadDocsOptions: "/applications/upload-docs-options",
    uploadId: "/applications/upload-id",
    workPatternType: "/applications/work-pattern-type",
  },
  auth: {
    createAccount: "/create-account",
    forgotPassword: "/forgot-password",
    login: "/login",
    resetPassword: "/reset-password",
    verifyAccount: "/verify-account",
  },
  employers: {
    addOrganization: "/employers/organizations/add-organization",
    cannotVerify: "/employers/organizations/cannot-verify",
    confirmation: "/employers/applications/confirmation",
    createAccount: "/employers/create-account",
    dashboard: "/employers/dashboard",
    newApplication: "/employers/applications/new-application",
    organizations: "/employers/organizations",
    review: "/employers/applications/review",
    status: "/employers/applications/status",
    success: "/employers/applications/success",
    verificationSuccess: "/employers/organizations/success",
    verifyContributions: "/employers/organizations/verify-contributions",
    welcome: "/employers/welcome",
  },
  external: {
    inLocoParentis:
      "https://www.dol.gov/agencies/whd/fact-sheets/28C-fmla-eldercare",
    massTaxConnect: "https://mtc.dor.state.ma.us/mtc/_/",
    massgov: {
      approvalTimeline:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-application-approval-timeline",
      benefitsCalculator:
        "https://calculator.digital.mass.gov/pfml/yourbenefits/",
      benefitsGuide: "https://www.mass.gov/how-pfml-benefits-are-calculated",
      benefitsGuide_aboutBondingLeave:
        "https://www.mass.gov/family-leave-bonding",
      benefitsGuide_aboutCaringLeave:
        "https://www.mass.gov/family-leave-caring",
      benefitsGuide_aboutMedicalLeave:
        "https://www.mass.gov/leave-types-medical",
      benefitsGuide_benefitsAmountDetails:
        "https://www.mass.gov/how-much-will-i-get-paid",
      benefitsGuide_seriousHealthCondition:
        "https://www.mass.gov/pfml-serious-health-condition",
      calculateHours: "https://www.mass.gov/pfml-calculate-hours",
      calculateReductions:
        "https://www.mass.gov/guides/what-to-expect-when-you-apply-for-paid-family-and-medical-leave-benefits#-calculate-any-reductions-",
      caregiverCertificationForm: "https://www.mass.gov/family-caring-form",
      caregiverRelationship:
        "https://www.mass.gov/family-caring-leave-relationships",
      caseCreationErrorGuide:
        "https://www.mass.gov/forms/apply-for-paid-leave-if-you-received-an-error",
      consentAgreement:
        "https://www.mass.gov/info-details/massachusetts-department-of-family-and-medical-leave-informed-consent-agreement",
      dfml: "https://www.mass.gov/orgs/department-of-family-and-medical-leave",
      dfmlRegulations:
        "https://www.mass.gov/regulations/458-CMR-200-family-and-medical-leave",
      employerAccount:
        "https://www.mass.gov/how-to/creating-an-employer-account-to-review-paid-family-and-medical-leave-pfml-applications",
      employerReimbursements:
        "https://www.mass.gov/forms/request-reimbursement-for-your-employees-paid-leave-benefits",
      employersGuide:
        "https://www.mass.gov/info-details/employer-paid-leave-application-review-guide",
      federalEmployerIdNumber:
        "https://www.mass.gov/info-details/finding-your-employers-federal-employer-identification-number-fein",
      feedbackClaimant: "https://www.mass.gov/paidleave-claimant-feedback",
      feedbackEmployer: "https://www.mass.gov/paidleave-employer-feedback",
      healthcareProviderForm: "https://www.mass.gov/medical-leave-form",
      howToApplyPaidLeave:
        "https://www.mass.gov/guides/what-is-paid-family-and-medical-leave#-how-can-i-apply-for-paid-leave-massachusetts-benefits",
      identityProof:
        "https://www.mass.gov/info-details/identity-proof-for-paid-leave",
      index:
        "https://www.mass.gov/topics/paid-family-and-medical-leave-in-massachusetts",
      informedConsent: "https://www.mass.gov/paidleave-informedconsent",
      intermittentLeaveGuide:
        "https://www.mass.gov/intermittent-leave-instructions",
      mailFaxInstructions: "https://www.mass.gov/pfmlsubmitinfo",
      manageApprovedApplication:
        "https://www.mass.gov/info-details/managing-your-approved-paid-family-or-medical-leave-claim",
      medicalBonding: "https://www.mass.gov/pfml-medical-bonding",
      multipleBirths: "https://www.mass.gov/pfml-multiple-births",
      onlineAppealsForm:
        "https://www.mass.gov/forms/request-an-appeal-for-paid-family-and-medical-leave",
      paidLeave:
        "https://www.mass.gov/topics/paid-family-and-medical-leave-in-massachusetts",
      pfml: "https://www.mass.gov/pfml",
      privacyPolicy: "https://www.mass.gov/privacypolicy",
      proofOfBirthOrPlacement:
        "https://www.mass.gov/info-details/required-documents-for-your-paid-family-and-medical-leave-pfml-application#for-family-leave-applications-",
      reductionsEmployerBenefits:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefit-reductions#earned-time-off-through-your-employer-",
      reductionsOverview:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefit-reductions",
      requestAnAppealForPFML:
        "https://www.mass.gov/how-to/appealing-a-denied-paid-family-or-medical-leave-claim",
      schedulingLeaveGuide:
        "https://www.mass.gov/info-details/intermittent-and-reduced-leave-schedules",
      sevenDayWaitingPeriodInfo: "https://www.mass.gov/getting-paid",
      taxGuide:
        "https://www.mass.gov/service-details/find-a-tax-professional-to-help-with-your-taxes",
      taxLiability: "https://www.mass.gov/pfml-tax-liability",
      timeline:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-application-approval-timeline",
      usingAccruedPaidLeave:
        "https://www.mass.gov/info-details/how-other-leave-and-benefits-can-affect-your-paid-family-and-medical-leave#paid-time-off-through-your-employer-",
      usingOtherLeave:
        "https://www.mass.gov/info-details/how-other-leave-and-benefits-can-affect-your-paid-family-and-medical-leave",
      verifyEmployer: "https://www.mass.gov/pfml-verify-employer",
      whenCanIUsePFML:
        "https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-fact-sheet#when-can-i-use-pfml?-",
      whichPaidLeave: "https://www.mass.gov/which-paid-leave",
      zeroBalanceEmployer: "https://www.mass.gov/pfml-zero-balance-employer",
    },
    puertoRicanBirthCertificate: "https://prfaa.pr.gov/faqs",
    workVisa:
      "https://travel.state.gov/content/travel/en/us-visas/employment/temporary-worker-visas.html",
  },
  index: "/",
  twoFactor: {
    smsConfirm: "/two-factor/sms/confirm",
    smsIndex: "/two-factor/sms",
    smsSetup: "/two-factor/sms/setup",
    smsVerify: "/two-factor/sms/verify",
  },
  user: {
    consentToDataSharing: "/user/consent-to-data-sharing",
    convert: "/user/convert-to-employer",
    settings: "/user/settings",
  },
};

/**
 * @param url - path, with or without query param
 */
export const isEmployersRoute = (url: string) => {
  return Object.values(routes.employers).includes(getRouteFromUrl(url));
};

/**
 * @param url - path, with or without query param
 */
export const isApplicationsRoute = (url: string) => {
  return Object.values(routes.applications).includes(getRouteFromUrl(url));
};

/**
 * @param url - path, with or without query param
 */
const getRouteFromUrl = (url: string) => {
  let route = url;
  const queryStringIndex = route.indexOf("?");

  if (queryStringIndex !== -1) {
    route = route.substring(0, queryStringIndex);
  }

  if (route.endsWith("/")) {
    route = route.substring(0, route.length - 1);
  }

  return route;
};

export default routes;
