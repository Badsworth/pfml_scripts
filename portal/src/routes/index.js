/* eslint sort-keys: ["error", "asc"] */
/**
 * @file Routes to various pages in the application are defined here rather than being
 * hard-coded into various files.
 */
const routes = {
  api: {
    claims: "/applications",
    users: "/users",
  },
  applications: "/applications",
  auth: {
    createAccount: "/create-account",
    forgotPassword: "/forgot-password",
    login: "/login",
    resetPassword: "/reset-password",
    verifyAccount: "/verify-account",
  },
  claims: {
    averageWorkHours: "/claims/average-work-hours",
    checklist: "/claims/checklist",
    confirm: "/claims/confirm",
    dashboard: "/",
    dateOfBirth: "/claims/date-of-birth",
    duration: "/claims/duration",
    employerBenefitDetails: "/claims/employer-benefit-details",
    employerBenefits: "/claims/employer-benefits",
    employmentStatus: "/claims/employment-status",
    leaveDates: "/claims/leave-dates",
    leaveReason: "/claims/leave-reason",
    name: "/claims/name",
    notifiedEmployer: "/claims/notified-employer",
    otherIncomes: "/claims/other-incomes",
    otherIncomesDetails: "/claims/other-incomes-details",
    paymentMethod: "/claims/payment-method",
    previousLeaves: "/claims/previous-leaves",
    previousLeavesDetails: "/claims/previous-leaves-details",
    reasonPregnancy: "/claims/reason-pregnancy",
    review: "/claims/review",
    ssn: "/claims/ssn",
    stateId: "/claims/state-id",
    success: "/claims/success",
    // For routes that don't have a page to point to yet, we can route them
    // to a placeholder page. This allows us to search our code for routes.claims.todo,
    // which is less confusing than seeing routes.claims.checklist
    todo: "/claims/checklist",
    uploadHealthcareForm: "/claims/upload-healthcare-form",
    uploadOtherId: "/claims/upload-other-id",
    uploadStateId: "/claims/upload-state-id",
  },
  external: {
    massgov: {
      healthcareProviderForm:
        "https://www.mass.gov/how-to/download-a-healthcare-provider-form",
      informedConsent: "https://www.mass.gov/paidleave-informedconsent",
      privacyPolicy: "https://www.mass.gov/privacypolicy",
    },
  },
  home: "/",
  user: {
    consentToDataSharing: "/user/consent-to-data-sharing",
  },
};

export default routes;
