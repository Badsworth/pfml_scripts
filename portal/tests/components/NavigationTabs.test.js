import { render, screen } from "@testing-library/react";
import NavigationTabs from "../../src/components/NavigationTabs";
import React from "react";
import routes from "../../src/routes";

const renderTabs = (customProps) => {
  const props = {
    tabs: [
      {
        label: "Employer Welcome",
        href: routes.employers.welcome,
      },
      {
        label: "Employer Organizations",
        href: routes.employers.organizations,
      },
    ],
    activePath: routes.employers.welcome,
    ...customProps,
  };
  return render(<NavigationTabs {...props} />);
};

describe("NavigationTabs", () => {
  it("renders the component", () => {
    const { container } = renderTabs();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("shows a link for each route", () => {
    renderTabs();
    expect(screen.getAllByRole("link")).toHaveLength(2);
    expect(
      screen.getByRole("link", { name: "Employer Welcome" })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("link", { name: "Employer Organizations" })
    ).toBeInTheDocument();
  });

  it("gives special styling to the active tab", () => {
    renderTabs();
    expect(
      screen.getByRole("link", { name: "Employer Welcome", current: "page" })
    ).toHaveClass("border-primary", "text-primary");
    expect(
      screen.getByRole("link", { name: "Employer Organizations" })
    ).not.toHaveClass("border-primary", "text-primary");
  });

  it("ignores queries in links given ignoreQueries prop", () => {
    renderTabs({
      ignoreQueries: true,
      activePath: routes.applications.status.claim,
      tabs: [
        {
          label: "Claim Detail",
          href: `${routes.applications.status.claim}?absence_case_id=NTN-123-213-1234`,
        },
        {
          label: "Payments",
          href: `${routes.applications.status.payments}?absence_case_id=NTN-123-213-1234`,
        },
      ],
    });
    expect(screen.getAllByRole("link")).toHaveLength(2);
    expect(
      screen.getByRole("link", { name: "Claim Detail", current: "page" })
    ).toHaveClass("border-primary", "text-primary");
    expect(screen.getByRole("link", { name: "Payments" })).not.toHaveClass(
      "border-primary",
      "text-primary"
    );
  });
});
