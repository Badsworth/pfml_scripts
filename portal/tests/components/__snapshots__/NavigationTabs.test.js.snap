// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationTabs renders the component 1`] = `
<nav
  class="border-bottom border-base-lighter margin-bottom-5 display-flex width-full"
>
  <a
    aria-current="page"
    class="display-inline-block padding-y-2 margin-bottom-neg-1px text-no-underline width-card border-primary border-bottom-05 text-bold text-primary"
    href="/employers/welcome"
  >
    Employer Welcome
  </a>
  <a
    class="display-inline-block padding-y-2 margin-bottom-neg-1px text-no-underline width-card text-base-dark hover:text-primary hover:text-underline"
    href="/employers/organizations"
  >
    Employer Organizations
  </a>
</nav>
`;
